### PR TITLE
improvements for support tools:

### PR DESCRIPTION
**Pull request by regehr**
a new combined static/dynamic profile sort for cache_dump

cache_dump can "weaken" dataflow facts-- currently, drop known bits
that are not needed to make a given optimization go

synthesis support + timeouts for cache_infer

code from Raimondas for souper2llvm

new tool called cache_import that takes a dumped text file full of
optimizations and loads it into Redis

### CODE REVIEW

File: utils/souper2llvm.in [hunk 1]
```diff
@@ -379,13 +379,13 @@ def rewriteOps(inst, newregs, ops):
...
     # global declarations: dummy store, intrinsics
     print "target datalayout = \"e-m:e-i64:64-f80:128-n8:16:32:64-S128\"\n"
-    print "@dummy1 = global i32 0, align 4"
-    print "@dummy2 = global i32 0, align 4"
-    print "@dummy3 = global i32 0, align 4"
-    print
+    #print "@dummy1 = global i32 0, align 4"
+    #print "@dummy2 = global i32 0, align 4"
+    #print "@dummy3 = global i32 0, align 4"
+    #print
```
Comment (rsas) [inline utils/souper2llvm.in]: This is required for `printFuncFooter()` (that is not being used anymore), so we should probably delete it.

File: utils/souper2llvm.in [hunk 2]
```diff
@@ -434,16 +434,20 @@ def printFuncFooter():
...
+    print "}"
+
 lines = readOpt()
 insts = parseInsts(lines)
 insts = propagateArgNames(insts)
 insts = groupPhis(insts)
 insts = rewriteRegs(insts)
 
-assert not needsResult, "result inst found"
+#assert not needsResult, "result inst found"
```
Comment (rsas) [inline utils/souper2llvm.in]: Same here.

**Comment by regehr** [comment]
Thanks-- updated.

**Comment by rsas** [comment]
LGTM

**Pull request merged**